README.md

```diff
@@ -34,16 +36,344 @@ version can be found on in the maven repository for
 </dependency>
 ```
 
-### Getting Started
+### Javadocs
 There are canonical examples defined in the class definition Javadoc of the client packages.
 
 Documentation can be found at the [Java Client
 Github Project Page](http://prometheus.github.io/client_java).
 
+
+## Instrumenting
+
+Four types of metrics are offered: Counter, Gauge, Summary and Histogram.
+See the documentation on [metric types](http://prometheus.io/docs/concepts/metric_types/)
+and [instrumentation best practices](http://prometheus.io/docs/practices/instrumentation/#counter-vs.-gauge,-summary-vs.-histogram)
+on how to use them.
+
+### Counter
+
+Counters go up, and reset when the process restarts.
+
+
+```java
+import io.prometheus.client.Counter;
+class YourClass {
+  static final Counter requests = Counter.build()
+     .name("requests_total").help("Total requests.").register();
+  
+  void processRequest() {
+    requests.inc();
+    // Your code here.
+  }
+}
+```
+
+### Gauge
+
+Gauges can go up and down.
+
+```java
+class YourClass {
+  static final Gauge inprogressRequests = Gauge.build()
+     .name("inprogress_requests").help("Inprogress requests.").register();
+  
+  void processRequest() {
+    inprogressRequest.inc();
+    // Your code here.
+    inprogressRequest.dec();
+  }
+}
+```
+
+There are utilities for common use cases:
+
+```java
+gauge.setToCurrentTime(); // Set to current unixtime.
+```
+
+As an advanced use case, a `Gauge` can also take its value from a callback. Keep in mind that the default `inc()`, `dec()` and `set()`
```

Comment:
As already discussed, link to the javadoc here rather than providing an example.